### PR TITLE
Bump KFP to 0.2.5

### PR DESCRIPTION
**Pull request by Bobgy**
**Which issue is resolved by this Pull Request:**
Resolves https://github.com/kubeflow/manifests/issues/984
Part of https://github.com/kubeflow/kubeflow/issues/4929

**Description of your changes:**
Bump KFP to 0.2.5

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate-changed-only`
    3. `make test`

### CODE REVIEW

**Comment by Bobgy** [comment]
I cannot build manifest tests. I got the following error when running `make generate-changed-only`:
```
INFO|2020-04-08T16:44:34|../hack/generate_tests.py|47| Creating directory /Users/gongyuan/kfp/manifests/tests/tests/legacy_kustomizations/centraldashboard/test_data/expected
Error: merging from generator &{0xc00003a540 0xc00011abb8 {map[] map[] false} {{ parameters merge {[] [] [params_0.env] }}}}: cannot find resource with id ~G_v1_ConfigMap|~X|parameters to replace
Traceback (most recent call last):
  File "../hack/generate_tests.py", line 188, in <module>
    run_kustomize_build(repo_root, full_dir)
  File "../hack/generate_tests.py", line 52, in run_kustomize_build
    package_dir))
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['kustomize', 'build', '--load_restrictor', 'none', '-o', u'/Users/gongyuan/kfp/manifests/tests/tests/legacy_kustomizations/centraldashboard/test_data/expected']' returned non-zero exit status 1
```
@richardsliu @jlewi any clue what might be wrong?

I'm using kustomize v2.1.0.

**Comment by richardsliu** [comment]
@Bobgy Please take a look at this issue for the fix: https://github.com/kubeflow/metadata/issues/202

**Comment by Bobgy** [comment]
@richardsliu thanks, so we use kustomize v3.
Okay, I will fix the manifest

**Comment by Bobgy** [comment]
@richardsliu I updated and fixed the test. PTAL again

**Comment by Bobgy** [comment]
Actually, there are still some problems when KFP runs.
* tfx example cannot get `metadata-grpc-configmap` in kubeflow namespace, because it is actually named `metadata-metadata-grpc-configmap-2dd6h7mhg6` in kubeflow. (common prefix metadata- added the duplicate prefix, and configmap generator added the hash, can we drop them?)

**Comment by Bobgy** [comment]
@zhenghuiwang @jlewi please review the metadata grpc configmap change

File: tests/legacy_kustomizations/metadata/kustomization.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ configMapGenerator:
 - behavior: merge
   envs:
   - params_1.env
-  name: metadata-grpc-configmap
+  name: grpc-configmap
```
Comment (Bobgy) [inline tests/legacy_kustomizations/metadata/kustomization.yaml:20]: Is this expected? I had to manually fix this file.

**Comment by Bobgy** [comment]
Added one more change to fix visualization server GCP permission.

**Comment by Bobgy** [comment]
Okay, I think I got all basic features integrated by default.
* all samples (including a tfx pipeline)
* tfx visualization
* tensorboard
* pipeline run/retry

**Comment by Bobgy** [comment]
This is ready for complete review

File: tests/legacy_kustomizations/metadata/kustomization.yaml (hunk 1 above)
Comment (zhenghuiwang) [reply tests/legacy_kustomizations/metadata/kustomization.yaml:20]: This doesn't seem to be right. I remember the name has to be `metadata-grpc-configmap`.
cc @dushyanthsc

**Comment by zhenghuiwang** [comment]
@Bobgy If you have verified TFX works then I think it is good to be merged.

/assign @dushyanthsc

**Comment by zhenghuiwang** [comment]
/lgtm

It correctly generated `configmap/metadata-grpc-configmap` with kustomize 2.0.3

**Comment by Bobgy** [comment]
Thanks @richardsliu @zhenghuiwang for review!

Note, metadata kustomization.yaml specified commonPrefix as `metadata-`, so I needed to drop each occurences of extra `metadata-` from `grpc-configmap`.

**Pull request merged**